update

```diff
@@ -59,19 +65,35 @@ do_fixedref()
     )
 }
 
-checkout_ffmpeg=do_gitmaster
-#checkout_fribidi=do_releasetag
-checkout_libass=do_gitmaster
-checkout_mpv=do_gitmaster
+# args: $1: project name, $2: release/master [,$3: non-empty to ignore the config file]
+checkout()
+{
+    local arg="$2"
+    if [ -z "$3" ] && [ -d config/ ] && [ -f config/branch-$1 ]; then
+        arg="$(cat config/branch-$1)"
+    fi
+
+    case "$arg" in
+        master)  do_gitmaster $1;;
+        *)       do_releasetag $1;;
+    esac
+}
+
+# fallback targets: release/master if no config file
+checkout_ffmpeg=master
+#checkout_fribidi=release
```

Comment:
Let's just remove any mention of fribidi.

Reply:
Actually, as someone who can read (some) RTL text, I can attest it's really important for RTL text, and without it RTL subs are displayed wrongly.

Unless you you feel strongly about removing it, I think it'd be useful to keep it.

Reply:
Fribidi isn't optional in libass, but most time the system version is fine.

Reply:
Gotcha. Sure, I'll remove it then.

Reply:
> Gotcha. Sure, I'll remove it then.

Actually, I have plans to keep supporting fribidi in a formal way but keeping it disabled by default in mpv-build, following wm4'a comment.

So I'll leave it commented out for now, and if my plan doesn't pan out then we can remove the comments later.

Reply:
Where is a non-system fribidi needed? Is it more important than other dependencies that are not included in mpv-build? My understanding is that this was added years ago with little explanation (4d6f1256116aa785992e2ed1df78c1d9932ceb3b) for someone's special case, but actually building it has been disabled (b38387044631aa433ae1612efbe9b1ed1e264b51) for most of that time with no one complaining.

Reply:
> Where is a non-system fribidi needed?

I don't know. I based it on wm4's comment which I interpreted as there are system where the global fribidi is not good enough.

> Is it more important than other dependencies that are not included in mpv-build?

There are a lot of them and I didn't line them up with priorities, but based on the fact that fribidi was there and others were not, seems to suggest that it might. But that's lots of ifs and buts.

However, it's definitely easier to support because we already have the config and build scripts for it, and we don't for others.

In general, my plan is to structure it in a way that it's easy to add more. The framework for config/build is already there to be extended relatively easily, but the update script is the only part which is a bit hardcoded towards a specific set of libs.

So moving the use-foo-bar scripts into a single config file would be one step in making it easy to extend without inflating the scripts count, and the update script hopefully can be maded similarly simple to extend with new libs.

Fribidi is just one which is both relatively easy to support and can show off the capabilities of having something formally supported yet disabled by default (where disabled here means to use the system libs).